plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -269,6 +270,11 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
         tailPlc (SomeConstantOfArg uniA (SomeConstantOfRes uniListA xs)) = case xs of
             _ : xs' -> EvaluationSuccess . SomeConstantOfArg uniA $ SomeConstantOfRes uniListA xs'
             _       -> EvaluationFailure
+    toBuiltinMeaning ChooseList = makeBuiltinMeaning choosePlc mempty where
+        choosePlc :: Opaque term b -> Opaque term b -> SomeConstantOf uni [] '[a] -> EvaluationResult (Opaque term b)
+        choosePlc a b (SomeConstantOfArg _ (SomeConstantOfRes _ xs)) = case xs of
+            []    -> EvaluationSuccess a
+            _ : _ -> EvaluationSuccess b
```

Comment:
You don't need `EvaluationResult` if you always return `EvaluationSuccess`. I.e. you can drop all of that.

Reply:
I don't know how to do that :thinking: I think I tried to write it like the other functions and it didn't work for some reason?